spec/models/patient_spec.rb

```diff
@@ -199,6 +201,89 @@ def refresh_views
 
         expect(described_class.ltfu_as_of(Time.current)).not_to include(not_ltfu_patient)
       end
+
+      describe "timezone-specific boundaries" do
+        around do |example|
+          Time.use_zone(reporting_timezone) do
+            Groupdate.time_zone = reporting_timezone
+            example.run
+            Groupdate.time_zone = nil
+          end
+        end
+
+        it "bp cutoffs for a year ago" do
+          # For any provided date in June in the local timezone, the LTFU BP cutoff is the end of June 30 of the
+          # previous year in the local timezone.
+          #
+          # Eg. For any date provided in June 2021, the cutoff is the June 30-Jul 1 boundary of 2020
+
+          long_ago = 5.years.ago
+          under_a_year_ago = Time.zone.local(2020, 7, 1, 0, 0, 1) # Beginning of July 1 2020 in local timezone
```

Comment:
You could use iso8601 strings here ? It would make this setup much more clear, and it looks like parse works fine with a little whitespace to make it more readable:

comparison:


```ruby
[52] pry(main)> under_a_year_ago = Time.zone.local(2020, 7, 1, 0, 0, 1)
=> Wed, 01 Jul 2020 00:00:01 IST +05:30
[53] pry(main)> under_a_year_ago = Time.zone.parse("2020-07-01 T00:00:01 +05:30")
=> Wed, 01 Jul 2020 00:00:01 IST +05:30
```